docs/windowing.md

```diff
@@ -0,0 +1,359 @@
+# -> Timestamp=101, value=1 -> emit {"start": 0, "end": 10000, "value": 2} 
+# -> Timestamp=102, value=1 -> emit {"start": 0, "end": 10000, "value": 3} 
+```
+
+`current()` mode may be used when you need to react on changes quickly, because the application doesn't need to wait until the window is closed. 
+But you will likely get duplicated values for each window interval
+
+
+2. `final()` - results are emitted only after the window is expired.
+
+This mode makes the application to wait until the maximum observed timestamp for topic partition passes the window end before emitting.
+
+Example:
+```python
+from datetime import timedelta
+
+
+sdf = app.dataframe(...)
+
+# Calculate a sum of values over a window of 10 seconds and emit results only when they are final
+sdf = sdf.tumbling_window(timedelta(seconds=10)).sum().current()
+
+
+# -> Timestamp=100, value=1   -> emit nothing (the window is not closed yet) 
+# -> Timestamp=101, value=1   -> emit nothing (the window is not closed yet) 
+# -> Timestamp=10001, value=1 -> emit {"start": 0, "end": 10000, "value": 2}, because the time has progressed beyond the window end. 
+```
+
+Emitting final results provides unique and complete values per window interval, but it adds some latency.
+Also, specifying a grace period via `grace_ms` will increase the latency, because the window now needs to wait for potential out-of-order events.
+
+`final()` mode is suitable for use cases when some latency is allowed, but the emitted events must be complete and unique.
+
+>***NOTE:*** Windows can be closed only by the records with the **same** message key.
+> If some message keys appear irregularly in the stream, the latest windows 
+> can remain unprocessed until the message with the same key is received.
+
+
+## Supported Aggregations
+
+Currently, windows support the following aggregation functions:
+
+- [`reduce()`](https://github.com/quixio/quix-streams/blob/main/docs/api-reference/quixstreams.md#fixedtimewindowdefinitionreduce) - to perform custom aggregations using "reducer" and "initializer" functions
+- [`min()`](https://github.com/quixio/quix-streams/blob/main/docs/api-reference/quixstreams.md#fixedtimewindowdefinitionmin) - to get a minimum value within a window
+- [`max()`](https://github.com/quixio/quix-streams/blob/main/docs/api-reference/quixstreams.md#fixedtimewindowdefinitionmax) -  to get a maximum value within a window
+- [`mean()`](https://github.com/quixio/quix-streams/blob/main/docs/api-reference/quixstreams.md#fixedtimewindowdefinitionmean) - to get a mean value within a window 
+- [`sum()`](https://github.com/quixio/quix-streams/blob/main/docs/api-reference/quixstreams.md#fixedtimewindowdefinitionsum) - to sum values within a window 
+- [`count()`](https://github.com/quixio/quix-streams/blob/main/docs/api-reference/quixstreams.md#fixedtimewindowdefinitioncount) - to count values within a window 
+
+
+## Examples
+
+### Calculate the sum of a message field over a tumbling window
```

Comment:
I am thinking, that this simple example would be nice at the beginning of the page as well. I am the sort of person who would just want to start with an example and then dig dipper into the docs and always appreciate something like this example at the beginning,